### PR TITLE
Add register name when register the class object

### PR DESCRIPTION
**Pull request by fengwuyao**
When register the class object:
- Generate the register name for the object.
- By default, the register name is "Custom>className"
- Construct a mapping between class object and register name.
- Add a getRegisterName() function to get registered name in mapping. If not registered, then return the name of input class.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

File: tfjs-core/src/serialization.ts [hunk 1]
```diff
@@ -151,15 +162,63 @@ export class SerializationMap {
...
+ * console.log(tf.serialization.GLOBALCUSTOMNAMES.get(MyCustomLayer));
  * ```
  *
  * @param cls The class to be registered. It must have a public static member
  *   called `className` defined and the value must be a non-empty string.
- *
+ * @param pkg The pakcage name that this class belongs to. This used to define
+ *     the key in GlobalCustomObject. If not defined, it defaults to `Custom`.
+ * @param name The name that user specified. It defaults to the actual name of
+ *     the class.
```
Comment (mattsoulanille) [inline tfjs-core/src/serialization.ts:217]:
```suggestion
 *     the class as specified by its static `className` property.
```

File: tfjs-core/src/serialization.ts [hunk 2]
```diff
@@ -54,6 +54,17 @@ export declare type SerializableConstructor<T extends Serializable> = {
...
     (cls: SerializableConstructor<T>, config: ConfigDict) => T;
 
+// export const GLOBALCUSTOMOBJECT:
+//     {[key: string]: SerializableConstructor<Serializable>} = {};
+export const GLOBALCUSTOMOBJECT =
+    new Map<string, SerializableConstructor<Serializable>>();
+
+// export const GLOBALCUSTOMNAMES: {[key:
+// SerializableConstructor<Serializable>]: string} = {};
+export const GLOBALCUSTOMNAMES =
```
Comment (mattsoulanille) [inline tfjs-core/src/serialization.ts:64]: Do we need to export this?

File: tfjs-core/src/serialization.ts [hunk 3]
```diff
@@ -54,6 +54,17 @@ export declare type SerializableConstructor<T extends Serializable> = {
 export declare type FromConfigMethod<T extends Serializable> =
     (cls: SerializableConstructor<T>, config: ConfigDict) => T;
 
+// export const GLOBALCUSTOMOBJECT:
+//     {[key: string]: SerializableConstructor<Serializable>} = {};
+export const GLOBALCUSTOMOBJECT =
```
Comment (mattsoulanille) [inline tfjs-core/src/serialization.ts:59]: Same here, do we need to export this?

Comment (fengwuyao) [reply tfjs-core/src/serialization.ts]: Right, we do not need to export this.

File: tfjs-core/src/serialization.ts [hunk 4]
```diff
@@ -54,6 +54,13 @@ export declare type SerializableConstructor<T extends Serializable> = {
 export declare type FromConfigMethod<T extends Serializable> =
     (cls: SerializableConstructor<T>, config: ConfigDict) => T;
 
+const GLOBAL_CUSTOM_OBJECT =
```
Comment (pyu10055) [inline tfjs-core/src/serialization.ts:57]: add comments for explain the usages of these two global map

File: tfjs-core/src/serialization.ts [hunk 5]
```diff
@@ -173,5 +228,47 @@ export function registerClass<T extends Serializable>(
...
+
+  const className = name;
+  const registerName = pkg + '>' + className;
+
+  if (GLOBAL_CUSTOM_OBJECT.has(registerName)) {
+    throw new Error(
+        `The name "${registerName}" has already been registered with class: ${
+            GLOBAL_CUSTOM_OBJECT.get(registerName)}`);
+  }
+  if (GLOBAL_CUSTOM_NAMES.has(cls)) {
```
Comment (pyu10055) [inline tfjs-core/src/serialization.ts:247]: this class registration check can be do at the beginning of the method.

**Review by pyu10055 — COMMENTED** [review]
LGTM with nit

File: tfjs-core/src/serialization.ts (hunk 4 above)
Comment (fengwuyao) [reply tfjs-core/src/serialization.ts:74]: SG, added the explanation and example usage

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**